powder_reduction.py

```diff
@@ -23,111 +44,459 @@ def powder_reduction(sample='sample.nxs', calibration=None, lambda_binning=(0.7,
+                                                 calibration,
+                                                 **absorp)
+
+        # The following selection of groups depends on the loaded data for
+        # Sample, Vanadium and Empty instrument
+
+        focused = focused['group', 0:5].copy()
+
+        vana_red_focused.coords['detector_info'] = focused.coords['detector_info'].copy()
+
+        focused /= vana_red_focused
+
+        del vana_red_focused
+
+    # 6. Prepare final result (for plotting)
+    dmin, dmax, number_bins_d = dspacing_binning
+    focused_hist = sc.histogram(focused,
+                                sc.Variable(['d-spacing'],
+                                            values=np.linspace(dmin, dmax, number_bins_d),
+                                            unit=sc.units.angstrom))
+
+    # Replace NaN, inf, -inf values
+    replace_value = sc.Variable(value=0.)
+
+    for i in range(focused_hist.coords['group'].shape[0]):
+        values_to_check = sc.Variable(dims=['d-spacing'],
+                                      values=focused_hist['group', i].values)
+
+        variances_to_check = sc.Variable(dims=['d-spacing'],
+                                         values=focused_hist['group', i].variances)
+
+        values_no_nan_inf = sc.nan_to_num(values_to_check,
+                                          nan=replace_value,
+                                          posinf=replace_value,
+                                          neginf=replace_value)
+
+        variances_no_nan_inf = sc.nan_to_num(variances_to_check,
+                                             nan=replace_value,
+                                             posinf=replace_value,
+                                             neginf=replace_value)
+
+        focused_hist['group', i].values = values_no_nan_inf.values
+        focused_hist['group', i].variances = variances_no_nan_inf.values
+
+    # Sum final result for plotting
+    sample_sum = sc.sum(focused_hist, 'group')
+
+    return sample_sum
+
+
+# #################################################################
+# Process event data for Vanadium and Empty instrument
+# #################################################################
+def process_event_data(file, lambda_binning):
+    """
+    Load and reduce event data for Vanadium and Empty instrument
+
+    file: Nexus file to be loaded and its data reduced in this script
+
+    lambda_binning: lambda_min, lamba_max, number_of_bins
+
+    """
+
+    # load nexus file
+    event_data = sc.neutron.load(file,
+                                 load_pulse_times=False,
+                                 mantid_args={'LoadMonitors': True})
+
+    # ################################
+    # Monitor correction
+    # extract monitor
+    mon4_selected = event_data.attrs['monitor4'].values
+
+    # convert to wavelength
+    mon4_lambda = sc.neutron.convert(mon4_selected, 'tof', 'wavelength')
+
+    del mon4_selected
```

Comment:
Shorter?
```suggestion
    mon4_lambda = sc.neutron.convert(event_data.attrs['monitor4'].value, 'tof', 'wavelength')
```

Reply:
OK I'll merge the 2 lines here and for the Sample data as well